solr/core/src/java/org/apache/solr/search/ExtendedDismaxQParser.java

```diff
@@ -532,13 +531,11 @@ protected List<ValueSource> getMultiplicativeBoosts() throws SyntaxError {
     List<ValueSource> boosts = new ArrayList<>();
     if (config.hasMultiplicativeBoosts()) {
       for (String boostStr : config.multBoosts) {
-        if (boostStr == null || boostStr.length() == 0) continue;
-        Query boost = subQuery(boostStr, FunctionQParserPlugin.NAME).getQuery();
-        ValueSource vs;
-        if (boost instanceof FunctionQuery) {
-          vs = ((FunctionQuery) boost).getValueSource();
-        } else {
-          vs = new QueryValueSource(boost, 1.0f);
+        if (boostStr == null || boostStr.isEmpty()) continue;
+        ValueSource vs = subQuery(boostStr, FunctionQParserPlugin.NAME).parseAsValueSource();
+        // the default score should be 1, not 0
+        if (vs instanceof QueryValueSource qvs && qvs.getDefaultValue() == 0.0f) {
+          vs = new QueryValueSource(qvs.getQuery(), 1.0f);
```

Comment:
not pretty but only had to do this here, so maybe doesn't warrant anything more elegant